coffeecard/CoffeeCard.Library/Services/v2/PurchaseService.cs

```diff
@@ -152,7 +153,26 @@ public async Task<SinglePurchaseResponse> GetPurchase(int purchaseId, User user)
                     $"No purchase was found by Purchase Id: {purchaseId} and User Id: {user.Id}");
             }
 
-            var paymentDetails = await _mobilePayPaymentsService.GetPayment(Guid.Parse(purchase.TransactionId));
+            PaymentDetails paymentDetails;
+            if (purchase.PaymentType == null)
```

Comment:
From my understanding, this assumes that all purchases done prior to this pull request were done using mobilePay. This however is not correct, due to us having issued vouchers as part of the datamodel.  As such, I believe line 159 will result in unintended behaviour.

Reply:
addressed in latest commit